From: Martin Michlmayr <tbm@cyrius.com>
Date: Tue, 30 Jun 2015 14:22:49 -0400
Subject: Add support for Aarch64 (ARM64)

Add support for Aarch64, the 64-bit ARM architecture.

Bug-Debian: https://bugs.debian.org/790668
Applied-upstream: (in ioquake3) https://github.com/ioquake/ioq3/commit/ebb69f699cd1392cbe7a865f9f51dbbecdd99b59
---
 Makefile                  | 4 ++++
 code/qcommon/q_platform.h | 2 ++
 2 files changed, 6 insertions(+)

diff --git a/Makefile b/Makefile
index 9732cc6..d36feaf 100644
--- a/Makefile
+++ b/Makefile
@@ -227,6 +227,10 @@ ifeq ($(PLATFORM),linux)
   else
   ifeq ($(ARCH),s390x)
     LIB=lib64
+  else
+  ifeq ($(ARCH),aarch64)
+    LIB=lib64
+  endif
   endif
   endif
   endif
diff --git a/code/qcommon/q_platform.h b/code/qcommon/q_platform.h
index 0ce4c29..ed3bfbb 100644
--- a/code/qcommon/q_platform.h
+++ b/code/qcommon/q_platform.h
@@ -175,6 +175,8 @@ Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
 #define ARCH_STRING "sparc"
 #elif defined __arm__
 #define ARCH_STRING "arm"
+#elif defined __aarch64__
+#define ARCH_STRING "aarch64"
 #elif defined __cris__
 #define ARCH_STRING "cris"
 #elif defined __hppa__
